psakefile.ps1

```diff
@@ -0,0 +1,55 @@
+include "./psake-build-helpers.ps1"
+
+$tag = $(git tag -l --points-at HEAD)
+$revision = @{ $true = ""; $false = "local" }[$env:BUILD_BUILDNUMBER -ne $NULL]
+$suffix = @{ $true = ""; $false = "ci-$revision" }[$tag -ne $NULL -and $revision -ne "local"]
```

Comment:
What are these used for other than to check if the suffix is or is not empty?

Reply:
Just for setting the suffix and we use it to set the nuget package version when we push the tag.

Reply:
Where exactly are we using it to set the nuget package version?

Reply:
as per the pipeline tasks, a new package is published to nuget only when the master is tagged. Otherwise, every master commit will create a package in private feed. We are using it in the task Build-Solution.

I am going to rename $suffix to $release and make it a boolean.

Reply:
GitVersion creates a "BranchName" variable you can use and pass onto your build script.